faucet/src/main.rs

```diff
@@ -0,0 +1,150 @@
+    id: AccountId,
+    asset_amount: u64,
+    client: Arc<Mutex<Client<TonicRpcClient, SqliteStore>>>,
+}
+
+#[actix_web::main]
+async fn main() -> std::io::Result<()> {
+    let cli = Cli::parse();
+
+    miden_node_utils::logging::setup_logging().map_err(|e| {
+        io::Error::new(io::ErrorKind::Unsupported, format!("Failed to load logging: {}", e))
+    })?;
+
+    // // Load config
+    // let config: FaucetTopLevelConfig =
+    //     load_config(cli.config.as_path()).extract().map_err(|e| {
+    //         io::Error::new(
+    //             io::ErrorKind::InvalidData,
+    //             format!("Failed to load configuration file: {}", e),
+    //         )
+    //     })?;
+
+    // // Instantiate Miden client
+    // let mut client = utils::build_client(config.faucet.database_filepath.clone());
+
+    let mut client: Client<TonicRpcClient, SqliteStore>;
+    let config: FaucetTopLevelConfig;
+    let amount: u64;
+
+    // Create faucet account
+    let faucet_account = match &cli.command {
+        cli::Commands::Init(InitArgs {
+            asset_amount,
+            token_symbol,
+            decimals,
+            max_supply,
+            config: faucet_config,
+        }) => {
+            config = load_config(faucet_config.as_path()).extract().map_err(|e| {
+                io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    format!("Failed to load configuration file: {}", e),
+                )
+            })?;
+
+            client = utils::build_client(config.faucet.database_filepath.clone());
+
+            amount = *asset_amount;
+            utils::create_fungible_faucet(token_symbol, decimals, max_supply, &mut client).map_err(
+                |e| {
+                    io::Error::new(
+                        io::ErrorKind::InvalidData,
+                        format!("Failed to create faucet account: {}", e),
+                    )
+                },
+            )
+        },
+        cli::Commands::Import(ImportArgs {
+            asset_amount,
+            faucet_path,
+            config: faucet_config,
+        }) => {
+            config = load_config(faucet_config.as_path()).extract().map_err(|e| {
+                io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    format!("Failed to load configuration file: {}", e),
+                )
+            })?;
+
+            client = utils::build_client(config.faucet.database_filepath.clone());
+
+            amount = *asset_amount;
+            utils::import_fungible_faucet(faucet_path, &mut client).map_err(|e| {
+                io::Error::new(
+                    io::ErrorKind::InvalidData,
+                    format!("Failed to import faucet account: {}", e),
+                )
+            })
+        },
+    }?;
+
+    // Sync client
+    client.sync_state().await.map_err(|e| {
+        io::Error::new(io::ErrorKind::ConnectionRefused, format!("Failed to sync state: {e:?}"))
```

Comment:
nit: I don't think the client failing to sync implies a `ConnectionRefused` necessarily, but I'm not familiar with the enum so not sure if there is a better match